concepts/05 Widgets/Scheduler/040 Resources/020 Assign Appointments to Resources/Assign Appointments to Resources.md

```diff
@@ -75,64 +75,201 @@ To define resource kinds, assign an array of objects specifying them to the [res
     // ...
     export class AppComponent  {
         // ...
-        rooms = [
-            // Resource instances
-            { 
-                id: 1,              // Resource identifier
-                text: "Room101",    // Resource name
-                color: "red"        // Color for indicating appointments that use this resource
+        rooms = new DataSource({
+            store: {
+                type: "array",
+                data: [
+                    // Resource instances
+                    {
+                        id: 1,              // Resource identifier
+                        text: "Room101",    // Resource name
+                        color: "red"        // Color for indicating appointments that use this resource
+                    },
+                    { id: 2, text: "Room102", color: "green" },
+                    // ...
+                ]
             },
-            { id: 2, text: "Room102", color: "green" },
-            // ...
-        ];
-        teachers = [
-            // Resource instances
-            { guid: "6F96", name: "John Heart", clr: "yellow" },
-            { guid: "3F32", name: "Sandra Johnson", clr: "blue" },
+            paginate: false
+        });
+
+        teachers = new DataSource({
+            store: {
+                type: "array",
+                data: [
+                    // Resource instances
+                    { guid: "6F96", name: "John Heart", clr: "yellow" },
+                    { guid: "3F32", name: "Sandra Johnson", clr: "blue" },
+                    // ...
+                ]
+            },
+            paginate: false
+        });
+    }
+    @NgModule({
+        imports: [
             // ...
-        ];
-        resources = [ 
-            // Definition of the first resource kind 
-            {
-                dataSource: new DataSource({
+            DxSchedulerModule
+        ],
+        // ...
+    })
+
+    <!--HTML-->
+    <dx-scheduler ... >
+
+        <!-- Definition of the first resource kind -->
+        <dxi-resource
+            fieldExpr="roomId"      <!-- "roomId" is the data field in an appointment object that binds it to the resource -->
+            label="Room"            <!-- Label displayed for this resource kind in the appointment details form -->
+            [dataSource]="rooms">
+        </dxi-resource>
+
+        <!-- Definition of the second resource kind -->
+        <dxi-resource
+            fieldExpr="teacherId"
+            valueExpr="guid"            // Resource instance's field used instead of "id"
+            colorExpr="clr"             // Resource instance's field used instead of "color"
+            displayExpr="name"          // Resource instance's field used instead of "text"
```

Comment:
```suggestion
            valueExpr="guid"            <!-- Resource instance's field used instead of "id" -->
            colorExpr="clr"             <!-- Resource instance's field used instead of "color" -->
            displayExpr="name"       <!-- Resource instance's field used instead of "text" -->
```